cdap-common/src/main/java/co/cask/cdap/common/logging/LocationAwareWrapperLogger.java

```diff
@@ -79,36 +96,56 @@ public boolean isTraceEnabled(Marker marker) {
 
   @Override
   public void trace(Marker marker, String msg) {
-    if (isTraceEnabled(marker) && sampler.accept(msg, LocationAwareLogger.TRACE_INT)) {
-      log(marker, FQCN, LocationAwareLogger.TRACE_INT, msg, null, null);
+    try {
+      if (isTraceEnabled(marker) && beforeLog(msg, LocationAwareLogger.TRACE_INT)) {
+        log(marker, FQCN, LocationAwareLogger.TRACE_INT, msg, null, null);
+      }
+    } finally {
+      afterLog();
     }
   }
 
   @Override
   public void trace(Marker marker, String format, Object arg) {
-    if (isTraceEnabled(marker) && sampler.accept(format, LocationAwareLogger.TRACE_INT)) {
-      log(marker, FQCN, LocationAwareLogger.TRACE_INT, format, new Object[]{arg}, null);
+    try {
```

Comment:
this is highly repetitive. Seems that the before(), try/catch, after() can be in one place, namely in the log() method.